### ISSUE TITLE
CORS Exposed headers are no longer sent down as response headers

### ISSUE THREAD

**Issue by HasseNasse** [quarkusio/quarkus#34093]
### Describe the bug

After migrating from Quarkus v.2.16.7 we see that configured exposed headers are not being sent back as a response from the server. The effect of this is that web-based clients strip off e.g the 'Location' header from the response headers of the API call.

### Expected behavior

When we configure an API as follows:

```java
@POST
    @Produces(MediaType.APPLICATION_JSON)
    public Response hello() {
        URI location = UriBuilder
                .fromResource(GreetingResource.class)
                .path("{id}")
                .resolveTemplate("id", "my-greeting")
                .build();
        return Response
                .created(location)
                .entity("Hello")
                .build();
    }
```

with the following cors config:
```properties
quarkus.http.cors=true
quarkus.http.cors.origins=*
quarkus.http.cors.exposed-headers=Location
quarkus.http.cors.methods=GET,PUT,POST,DELETE,OPTIONS
```

and we call the API (on 2.16.7) we get the following pre-flight response:
<img width="1062" alt="image" src="https://github.com/quarkusio/quarkus/assets/10097082/3bd4dd40-e577-4892-a7ea-9bca46d4a94e">

and the actual POST response as follows:
<img width="1001" alt="image" src="https://github.com/quarkusio/quarkus/assets/10097082/5c291cdb-8a43-43ba-a3bf-16a619197edd">

### Actual behavior

When we configure an API as follows:

```java
@POST
    @Produces(MediaType.APPLICATION_JSON)
    public Response hello() {
        URI location = UriBuilder
                .fromResource(GreetingResource.class)
                .path("{id}")
                .resolveTemplate("id", "my-greeting")
                .build();
        return Response
                .created(location)
                .entity("Hello")
                .build();
    }
```

with the following cors config:
```properties
quarkus.http.cors=true
quarkus.http.cors.origins=*
quarkus.http.cors.exposed-headers=Location
quarkus.http.cors.methods=GET,PUT,POST,DELETE,OPTIONS
```

and we call the API (on 3.1.2) we get the following pre-flight response:
<img width="1012" alt="image" src="https://github.com/quarkusio/quarkus/assets/10097082/00fdb75a-e38f-491b-a54f-6f9ad295bf48">

and the actual POST response as follows:
<img width="1021" alt="image" src="https://github.com/quarkusio/quarkus/assets/10097082/2cc40b80-8263-4c1a-bdb8-d430daffc72a">

As can be seen in the last image of v3.1.2, the POST response headers don't contain the exposed-headers that we configured. This means that JS will strip away that header from the client.

### How to Reproduce?

https://github.com/hreem-it/cors-test

Repo contains 2 branches `quarkus-v2` and `quarkus-v3` 

### Output of `uname -a` or `ver`

Darwin Hassans-MacBook-Pro.local 22.5.0 Darwin Kernel Version 22.5.0: Mon Apr 24 20:52:24 PDT 2023; root:xnu-8796.121.2~5/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk version "17.0.3" 2022-04-19 OpenJDK Runtime Environment GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06) OpenJDK 64-Bit Server VM GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06, mixed mode, sharing)

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537) Maven home: /Users/hassannazar/dev/tools/apache-maven-3.8.4 Java version: 17.0.3, vendor: GraalVM Community, runtime: /Library/Java/JavaVirtualMachines/graalvm-ce-java17-22.1.0/Contents/Home Default locale: en_GB, platform encoding: UTF-8 OS name: "mac os x", version: "13.4", arch: "aarch64", family: "mac"

**Comment by sberyozkin** [comment]
@HasseNasse Thanks for reporting. It was fixed with #33923.
Closing as a duplicate

**Comment by HasseNasse** [comment]
@sberyozkin when can we expect it to be released?

**Comment by sberyozkin** [comment]
@HasseNasse 3.2.0.CR1 is expected on Wed 21st June, https://github.com/quarkusio/quarkus/wiki/Release-Planning
but it may take another few weeks for 3.1.3.Final be released, unfortunately this fix just missed 3.1.2.Final

**Comment by sberyozkin** [comment]
@HasseNasse Hope you'll have no problems moving to `3.2.0.CR1`
